### PR TITLE
Template doc and a few small fixes

### PR DESCRIPTION
**Pull request by aulemahal**
### Pull Request Checklist:
- [ ] This PR addresses an already opened issue (for bug fixes / features)
    - This PR fixes #xyz
- [x] (If applicable) Documentation has been added / updated (for bug fixes / features)
- [x] HISTORY.rst has been updated (with summary of main changes)
  - [x] Link to issue (:issue:`number`) and pull request (:pull:`number`) has been added

### What kind of change does this PR introduce?

* Return only requested variables with `to_dataset`. The deepcopy was not copying the relevant attribute for some reason.
* `clisops_subset` preloads `lon` and `lat` to avoid loading them multiple times during the process.
* Add an "indicator-only" template
* Minimal documentation for templates using a github-readable "readme.rst" and linking the latter directly in the RTD index.
* Add a PR check and comment in the contributing notes about the checking if changes work with the untested templates.

### Does this PR introduce a breaking change?
No

### Other information:
When someone clicks the link from the doc to the templates folder, they land on the "main" branch. I didn't find an easy way to generate a link that depends on the branch/version of the documentation being read. Instead, a warning in the readme instructs how to go back to the released version. I'm not sure how much of an issue this is.

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -4,6 +4,7 @@
 - [ ] This PR addresses an already opened issue (for bug fixes / features)
     - This PR fixes #xyz
 - [ ] (If applicable) Documentation has been added / updated (for bug fixes / features)
+- [ ] This PR does not seem to break the templates.
```
Comment (RondeauG) [inline .github/PULL_REQUEST_TEMPLATE.md:7]: I don't think that we actively run them. Could we test it more thoroughly ? If not, I don't know how useful that statement is.

**Review by RondeauG — APPROVED** [review]

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (aulemahal) [reply .github/PULL_REQUEST_TEMPLATE.md:7]: My idea was that (for example) if you change the API, you at least check that the template is not broken ? That's why I used "seem".

Testing the templates would be a bit awkward I think...

Comment (juliettelavoie) [reply .github/PULL_REQUEST_TEMPLATE.md:7]: like do you want people to run them locally or just look and feel that they aren't broken ?
maybe we could run them at least each time we have a release ?

Comment (RondeauG) [reply .github/PULL_REQUEST_TEMPLATE.md:7]: They can't be run locally as they currently are, sine many things (such as catalog paths) are just placeholders. I'm fine with "just looking", as long as we develop tests eventually.

File: templates/2-indicators_only/workflow2.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+            outfreq = ind.injected_parameters["freq"].replace("YS", "AS-JAN")
+            outnames = [cfatt["var_name"] for cfatt in ind.cf_attrs]
+            if not pcat.exists_in_cat(
+                id=dsid,
+                variable=outnames,
+                xrfreq=outfreq,
+                processing_level="indicators",
+            ):
+                to_compute.append((name, ind))
+
```
Comment (juliettelavoie) [inline templates/2-indicators_only/workflow2.py:61]: J'aime ça! Dans mon workflow, je perdais du temps à faire le calcul lazy et je regardais juste que ça existait avant de save to zarr. C'est mieux comme ça!

File: templates/readme.rst [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+Most of the templates are heavily commented in their python code as well as their configuration files, but a small summary of what each does is included here.
+
+.. warning::
+
+	The link above brings you to the development version of the templates. You might want to access a version specific to your installed xscen. Click the dropdown menu in the upper left corner that says "main" and navigate to "tags" and the specific version of interest.
+
+
+1 - Basic workflow with config
+------------------------------
+The archetypal xscen workflow that does every steps of a normal climate scenarisation project : extract, regrid, biasadjust, cleanup, rechunk, diagnostics, indicators, climatology, delta, ensembles. It is controlled from the ``config1.yml`` file. For each step, it will iterate over each member of the ensemble, thus creating many intermediate files before the final products.
```
Comment (juliettelavoie) [inline templates/readme.rst:15]: pas vraiment rapport avec cette PR, mais ça me fait penser qu'on voudrait peut-être développer une fonction xscen pour gérer la suppression des fichiers intermédiaires ?

**Review by juliettelavoie — APPROVED** [review]
good!

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (juliettelavoie) [reply .github/PULL_REQUEST_TEMPLATE.md:7]: ouais, mais en l'absence de tests pour l'instant,  je suggérais que Pascal et moi (qui avons les paths), on roule les templates avant de passer la version 6.0 mettons.

Comment (aulemahal) [reply .github/PULL_REQUEST_TEMPLATE.md:7]: I think I'll add a note in the release process. Manually testing at each PR would be burdensome, maybe that could be done only before publishing. And yes, we'd need to think on how these can be tested...

File: templates/readme.rst (hunk 3 above)
Comment (aulemahal) [reply templates/readme.rst:15]: Genre, qqch qui lit le `pcat` et enlève certains  `processing_level` ?

Comment (juliettelavoie) [reply templates/readme.rst:15]: ouais qui comprends qu'on est rendu 2 tâches plus loin ou peut-être qui regarde les inputs dans une config comme dans le template1. L'idée n'est pas complètement formée dans ma tête.

**Pull request merged**